addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3791,6 +4007,84 @@ export class OdooEditor extends EventTarget {
             this._onTabulationInTable(ev);
         }
     }
+    _onTableMenuTogglerClick(ev) {
+        const uiWrapper = closestElement(ev.target, '.o_table_ui');
+        uiWrapper.classList.toggle('o_open');
+        if (uiWrapper.classList.contains('o_column_ui')) {
+            const columnIndex = getColumnIndex(this._columnUiTarget);
+            const moveLeft = uiWrapper.querySelector('.o_move_left');
+            moveLeft.classList.toggle('o_hide', columnIndex === 0);
+            const moveRight = uiWrapper.querySelector('.o_move_right');
+            moveRight.classList.toggle('o_hide', columnIndex === [...this._columnUiTarget.parentElement.children].filter(child => child.nodeName === 'TD').length - 1);
```

Comment:
Are you sure the nodes returned by the `querySelector` will always be defined ? If no then you should check them before accessing `classList`. If yes then I wouldn't have put them in `const` just to use them once on the line below. Sometimes it makes sense because the variable name adds valuable information, but in this case it's the same as the class you're querying for... ^^' I fear the lines would be too long and look too complex however so I suggest you try it and if you think it's worse then the current code is fine. (Same below.)

Reply:
As a matter of fact I am, they're just elements of the UI. It just makes line 4018 very long then but otherwise it's ok with me.

Reply:
Done. Two lines became very long:
```js
uiWrapper.querySelector('.o_move_right').classList.toggle('o_hide', columnIndex === [...this._columnUiTarget.parentElement.children].filter(child => child.nodeName === 'TD').length - 1);
```
and
```js
uiWrapper.querySelector('.o_move_down').classList.toggle('o_hide', rowIndex === [...this._rowUiTarget.parentElement.children].filter(child => child.nodeName === 'TR').length - 1);
```
so I took the booleans out of them and into named constants:
```js
const shouldHideRight = columnIndex === [...this._columnUiTarget.parentElement.children].filter(child => child.nodeName === 'TD').length - 1;
uiWrapper.querySelector('.o_move_right').classList.toggle('o_hide', shouldHideRight);
```
and
```js
const shouldHideDown = rowIndex === [...this._rowUiTarget.parentElement.children].filter(child => child.nodeName === 'TR').length - 1;
uiWrapper.querySelector('.o_move_down').classList.toggle('o_hide', shouldHideDown);
```

The lines were long to begin with and remain so but it's at least a bit better.